### PR TITLE
Remove the Cmd+S/Ctrl+S keyboard shortcut from the code editor

### LINKED ISSUE #829: Cmd + S in the code editor throws an error
Using Cmd + S in the code editor follows a code path for saving github gists that is no longer working.

### PR DESCRIPTION
**Pull request by chriswhong**
Closes #829 

* Removes the Cmd+S/Ctrl+S keyboard shortcut from the code editor
* Removes the unused code paths for saving to github and saving to gist, and updates any code that previously called them.

### CODE REVIEW

File: src/core/data.js [hunk 1]
```diff
@@ -272,22 +272,5 @@ module.exports = function (context) {
     }
   };
 
-  data.save = function (cb) {
-    const type = context.data.get('type');
-    if (type === 'github') {
-      source.github.save(context, cb);
-    } else if (type === 'gist') {
-      source.gist.save(context, cb);
-    } else if (type === 'local') {
```
Comment (wboykinm) [inline src/core/data.js:281]: Interesting - did `local` ever work? Does that refer to in-memory or to a downloaded file?

**Review by danpat — APPROVED** [review]

File: src/panel/json.js [hunk 2]
```diff
@@ -20,16 +19,9 @@ module.exports = function (context) {
       const spaces = new Array(cm.getOption('indentUnit') + 1).join(' ');
       cm.replaceSelection(spaces, 'end', '+input');
     },
-    'Ctrl-S': saveAction,
-    'Cmd-S': saveAction,
```
Comment (wboykinm) [inline src/panel/json.js:24]: salute

File: src/core/data.js (hunk 1 above)
Comment (chriswhong) [reply src/core/data.js:281]: No idea, all of the file downloads have their own code path, so not sure what `type` local ever was.

File: src/panel/json.js (hunk 2 above)
Comment (chriswhong) [reply src/panel/json.js:24]: :)

**Pull request merged**